tests/mock/datastore.py

```diff
@@ -0,0 +1,128 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from bisect import bisect_left
+from contextlib import contextmanager
+import lmdb
+from threading import Lock
+
+from constant_sorrow.constants import MOCK_DB
+
+
+def mock_lmdb_open(db_path, map_size=10485760):
+    if db_path == MOCK_DB:
+        return MockEnvironment()
+    else:
+        return lmdb.Environment(db_path, map_size=map_size)
+
+
+class MockEnvironment:
+
+    def __init__(self):
+        self._storage = {}
+        self._lock = Lock()
+
+    @contextmanager
+    def begin(self, write=False):
+        with self._lock:
+            with MockTransaction(self, write=write) as tx:
+                yield tx
+
+
+class MockTransaction:
```

Comment:
Consider:

```suggestion
class MockDatabaseTransaction
```

Reply:
I named the classes based on the original ones from `lmdb` - `Environment`, `Transaction` and `Cursor`. I think there's no harm in leaving it like this.